app/src/main/java/com/mstudio/superheroarch/repository/RickAndMortyRepository.kt

```diff
@@ -29,12 +29,11 @@ class RickAndMortyRepository(
         }
     }
 
-    suspend fun fetchEpisodeDetails(episodeUrl: String): Result<EpisodeEntity>{
+    suspend fun fetchEpisodeDetails(episodeUrl: String): EpisodeEntity {
         return try {
-            val response = remoteDataSource.getEpisode(episodeUrl)
-            Result.success(response)
-        }catch (e: IOException){
-            Result.failure(e)
+            remoteDataSource.getEpisode(episodeUrl)
+        } catch (e: IOException) {
+            throw e
```

Comment:
catch + throw means nothing. You are catching the exception and throwing it again, this is useless. Remove the try catch.

Reply:
Solved in next commit